### PR TITLE
Add dissect.target[yara] as an optional dependency

### PR DESCRIPTION
**Pull request by Miauwkeru**

### CODE REVIEW

**Review by pyrco — CHANGES_REQUESTED** [review]
The testenv in tox.ini should use the `yara` instead of the `full` extra.

File: README.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ pip install dissect.target
 ```
 
 This module is also automatically installed if you install the `dissect` package.
+To enable `yara` with `dissect` install `dissect.target[yara]`.
```
Comment (Schamper) [inline README.md:16]:
```suggestion
If you wish to use the YARA plugin (`target-query -f yara`), you can install `dissect.target[yara]` instead.

This module is also automatically installed if you install the `dissect` package.
```

**Review by Schamper — CHANGES_REQUESTED** [review]
Maybe also nice to add a log message or re-raised `ImportError` that states to install `yara-python` in the YARA plugin file?

File: README.md [hunk 2]
```diff
@@ -12,6 +12,9 @@ please see [the documentation](https://docs.dissect.tools/en/latest/projects/dis
 pip install dissect.target
 ```
 
+If you wish to use the YARA plugin (`target-query -f yara`), you can install `dissect.target[yara]` to automatically 
+install the `yara-python` dependency.
+
 This module is also automatically installed if you install the `dissect` package.
```
Comment (Schamper) [inline README.md:18]:
```suggestion
This module is also automatically installed if you install the `dissect` package.

If you wish to use the YARA plugin (`target-query -f yara`), you can install `dissect.target[yara]` to automatically 
install the `yara-python` dependency.
```

I changed my mind :smile: with my previous suggestion it's unclear what "This module" refers to, so better to put this new sentence below.

**Review by Schamper — CHANGES_REQUESTED** [review]

**Review by pyrco — APPROVED** [review]

**Review by Schamper — APPROVED** [review]

**Pull request merged**